### PR TITLE
more chunking upload tests

### LINKED ISSUE #216: Expand acceptance testing for uploads
With https://github.com/owncloud/guests/pull/212 we already included some basic chunked upload testing.

However, this only uses a specific stile of chunking. As ownCloud currently supports chunking v1 and v2 - we need to expand the test cases to cover both upload parts

### LINKED ISSUE #278: API/Acceptance tests for Async Upload
test if the Async Upload owncloud/core#32414 works correctly with the guests app

### PR DESCRIPTION
**Pull request by individual-it**
## Description
more test cases for chunked uploads, incl. async upload

## Related Issue
fixes #216
fixes #278 

## Motivation and Context
more tests = more better

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Tests

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/apiGuests/guests.feature [hunk 1]
```diff
@@ -57,6 +69,58 @@ Feature: Guests
     Then as "guest@example.com" file "/tmp/myChunkedFile.txt" should exist
     And as "user0" file "/tmp/myChunkedFile.txt" should exist
 
+  @mailhog @issue-279
+  Scenario: A guest user can upload files
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature]: A guest user can upload files to a received share

File: tests/acceptance/features/apiGuests/guests.feature [hunk 2]
```diff
@@ -25,8 +25,20 @@ Feature: Guests
     Given as user "%admin%"
     And user "%admin%" has created guest user "guest" with email "guest@example.com"
     And the HTTP status code should be "201"
-    When user "guest@example.com" uploads file "textfile.txt" from the guests test data folder to "/myfile.txt" using the WebDAV API
+    When user "guest@example.com" uploads overwriting file "textfile.txt" from the guests test data folder to "/myfile.txt" with all mechanisms using the WebDAV API
+    Then the HTTP status code of all upload responses should be "401"
+    And the HTTP reason phrase of all upload responses should be "Unauthorized"
+    And as "guest@example.com" file "/textfile.txt" should not exist
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature]: The step code for this expects a 404 "not found", but a the guest is returning 401 "unauthorized".
I guess the step code needs to be a bit smarter, and/or the API query for a guest should be returning 404, or there is a bug in the step code?
https://drone.owncloud.com/owncloud/guests/398/90
```
    And as "guest@example.com" file "/textfile.txt" should not exist                                                                                                 # FeatureContext::asFileOrFolderShouldNotExist()
      Exception: file '/textfile.txt' expected to not exist (status code 401, expected 404) in /var/www/owncloud/tests/acceptance/features/bootstrap/WebDav.php:1194
```

File: tests/acceptance/features/apiGuests/guests.feature [hunk 3]
```diff
@@ -25,8 +25,20 @@ Feature: Guests
...
+    And as "user0" file "/textfile.txt" should not exist
+
+  Scenario: A guest user cannot upload files (async upload)
+    Given as user "%admin%"
+    And the administrator has enabled async operations
+    And user "%admin%" has created guest user "guest" with email "guest@example.com"
+    When user "guest@example.com" uploads file "textfile.txt" from the guests test data folder asynchronously to "/textfile.txt" in 3 chunks using the WebDAV API
     Then the HTTP status code should be "401"
+    And as "guest@example.com" file "/textfile.txt" should not exist
+    And as "user0" file "/textfile.txt" should not exist
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature]: Same 404/401 problem here as in the scenario above.

File: tests/acceptance/features/apiGuests/guests.feature (hunk 2 above)
Comment (individual-it) [reply tests/acceptance/features/apiGuests/guests.feature]: I guess the server code just fails earlier when using the guests app

**Comment by phil-davis** [comment]
@individual-it it would be good to rebase this, because I merged another refactoring PR underneath.
(even though you are lucky, and no conflicts)

File: tests/acceptance/features/apiGuests/guests.feature [hunk 4]
```diff
@@ -25,8 +25,20 @@ Feature: Guests
...
+    When user "guest@example.com" uploads overwriting file "textfile.txt" from the guests test data folder to "/myfile.txt" with all mechanisms using the WebDAV API
+    Then the HTTP status code of all upload responses should be "401"
+    And the HTTP reason phrase of all upload responses should be "Unauthorized"
+    And as "guest@example.com" file "/textfile.txt" should not exist
+    And as "user0" file "/textfile.txt" should not exist
+
+  Scenario: A guest user cannot upload files (async upload)
+    Given as user "%admin%"
+    And the administrator has enabled async operations
+    And user "%admin%" has created guest user "guest" with email "guest@example.com"
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature]: s/user "%admin%"/the administrator

File: tests/acceptance/features/apiGuests/guests.feature [hunk 5]
```diff
@@ -57,6 +69,58 @@ Feature: Guests
     Then as "guest@example.com" file "/tmp/myChunkedFile.txt" should exist
     And as "user0" file "/tmp/myChunkedFile.txt" should exist
 
+  @mailhog @issue-279
+  Scenario: A guest user can upload files
+    Given as user "%admin%"
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature]: not needed

File: tests/acceptance/features/apiGuests/guests.feature [hunk 6]
```diff
@@ -57,6 +69,58 @@ Feature: Guests
...
+    And user "user0" has been created with default attributes
+    And user "%admin%" has created guest user "guest" with email "guest@example.com"
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature]: s/user "%admin%"/the administrator

File: tests/acceptance/features/apiGuests/guests.feature [hunk 7]
```diff
@@ -57,6 +69,58 @@ Feature: Guests
...
+    """
+    This is a testfile.
+    
+    Cheers.
+    """
+
+  @mailhog
+  Scenario: A guest user can upload files (async upload)
+    Given the administrator has enabled async operations
+    And as user "%admin%"
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature]: not needed

File: tests/acceptance/features/apiGuests/guests.feature [hunk 8]
```diff
@@ -57,6 +69,58 @@ Feature: Guests
...
+    And user "user0" has been created with default attributes
+    And user "%admin%" has created guest user "guest" with email "guest@example.com"
```
Comment (phil-davis) [inline tests/acceptance/features/apiGuests/guests.feature]: s/user "%admin%"/the administrator

**Comment by individual-it** [comment]
@ownclouders rebase

**Pull request merged**